3p/iframe-transport-client.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {tryParseJson} from '../src/json';
+import {dev, user} from '../src/log';
+import {MessageType} from '../src/3p-frame-messaging';
+import {IframeMessagingClient} from './iframe-messaging-client';
+
+/** @private @const {string} */
+const TAG_ = 'iframe-transport-client';
+
+/**
+ * Receives event messages bound for this cross-domain iframe, from all
+ * creatives
+ */
+export class IframeTransportClient {
+
+  /** @param {!Window} win */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {?function(string,string)} */
+    this.listener_ = null;
+
+    /** @protected {!IframeMessagingClient} */
+    this.client_ = new IframeMessagingClient(win);
+    this.client_.setHostWindow(this.win_.parent);
+    this.client_.setSentinel(dev().assertString(
```

Comment:
I agree. We should probably change it at https://github.com/ampproject/amphtml/blob/master/3p/ampcontext.js#L103 also